### PR TITLE
list of todos

### PR DESCRIPTION
**Pull request by YevhenOstrovskiy**

### CODE REVIEW

**Comment by YevhenOstrovskiy** [comment]
[DEMO LINK](https://YevhenOstrovskiy.github.io/react_dynamic-list-of-todos/)

File: src/App.tsx [hunk 1]
```diff
@@ -21,14 +24,20 @@ export const App: React.FC = () => {
...
       </div>
-
-      <TodoModal />
+      { selectedTodo && (<TodoModal />)}
     </>
   );
 };
+
+export const App: React.FC = () => {
+  return (
```
Comment (olya-shyian) [inline src/App.tsx:38]: there is no need to write word `return` if you don'y have any logic, just jsx

rewrite to
`= () => (`

File: src/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -1,100 +1,87 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+import { TodoContext } from '../TodoContext';
+// import { Todo } from '../../types/Todo';
```
Comment (olya-shyian) [inline src/components/TodoList/TodoList.tsx:4]: please delete commented code

File: src/components/TodoList/TodoList.tsx [hunk 3]
```diff
@@ -1,100 +1,87 @@
...
-        <th>
-          <span className="icon">
-            <i className="fas fa-check" />
-          </span>
-        </th>
-        <th>Title</th>
-        <th> </th>
-      </tr>
-    </thead>
+type Props = {};
```
Comment (olya-shyian) [inline src/components/TodoList/TodoList.tsx:6]: if there is no Props, you can delete it

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -1,100 +1,87 @@
...
-);
+      <tbody>
+        {todos.map(todo => (
+          <tr
+            data-cy="todo"
+            className={
+              classNames(
+                {
+                  'has-background-info-light':
+                selectedTodo?.id === todo.id,
```
Comment (olya-shyian) [inline src/components/TodoList/TodoList.tsx:40]: please check formatting

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -1,100 +1,87 @@
...
+            <td className="is-vcentered is-expanded">
+              <p className={classNames({
+                'has-text-danger': !todo.completed,
+                'has-text-success': todo.completed,
+              })}
+              >
+                {todo.title}
+              </p>
+            </td>
+            <td className="has-text-right is-vcentered">
```
Comment (olya-shyian) [inline src/components/TodoList/TodoList.tsx:63]: don't forget about code style, please add a line between inner elements

File: src/components/TodoModal/TodoModal.tsx [hunk 6]
```diff
@@ -1,49 +1,76 @@
...
-          </header>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                onClick={() => {
+                  setShowModal(!showModal);
+                  setUser(null);
+                  setSelectedTodo(null);
```
Comment (olya-shyian) [inline src/components/TodoModal/TodoModal.tsx:40]: create a reset handler for that

File: src/components/TodoModal/TodoModal.tsx [hunk 7]
```diff
@@ -1,49 +1,76 @@
...
+            <div className="modal-card-body">
+              <p className="block" data-cy="modal-title">
+                {selectedTodo?.title}
+              </p>
 
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              <p className="block" data-cy="modal-user">
+                {/* <strong className="has-text-success">Done</strong> */}
```
Comment (olya-shyian) [inline src/components/TodoModal/TodoModal.tsx:51]: please delete commented code

File: src/components/TodoModal/TodoModal.tsx [hunk 8]
```diff
@@ -1,49 +1,76 @@
...
+                <strong
+                  className={classNames({
+                    'has-text-danger': !selectedTodo?.completed,
+                    'has-text-success': selectedTodo?.completed,
```
Comment (olya-shyian) [inline src/components/TodoModal/TodoModal.tsx:55]: please use destructuring for `selectedTodo` : {completed, id, title}

**Review by olya-shyian — CHANGES_REQUESTED** [review]
Very good job!💪

1. Please remove loader if there is no data (check the original working page)
image

File: src/App.tsx [hunk 9]
```diff
@@ -21,14 +24,21 @@ export const App: React.FC = () => {
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              { todos.length ? (<TodoList />
+              ) : (
+                query === '' && <Loader />
```
Comment (DanilWeda) [inline src/App.tsx:29]:
```suggestion
                !query && <Loader />
```
as alternative.

Comment (DanilWeda) [inline src/App.tsx:29]: Need change. If the query is empty I see the loader, but I need to see an empty result.

File: src/components/TodoContext/TodoContext.tsx [hunk 10]
```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { getTodos, getUser } from '../../api';
+import { User } from '../../types/User';
+import { Context } from '../../types/Context';
+import { Todo } from '../../types/Todo';
+import { Status } from '../../types/Status';
+import { filterData } from '../services';
+
+type Props = {
```
Comment (DanilWeda) [inline src/components/TodoContext/TodoContext.tsx:9]: use interface prefer. 
Types have unique features but the interface is best for scaling.

File: src/components/TodoFilter/TodoFilter.tsx [hunk 11]
```diff
@@ -1,34 +1,57 @@
...
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        { query && (
```
Comment (DanilWeda) [inline src/components/TodoFilter/TodoFilter.tsx:43]:
```suggestion
        { !!query && (
```
in rendering with &&  need to convert the value to boolean.

File: src/components/TodoList/TodoList.tsx [hunk 12]
```diff
@@ -1,100 +1,81 @@
...
+              <button
+                data-cy="selectButton"
+                className="button"
+                type="button"
+                onClick={() => {
+                  setSelectedTodo(todo);
+                  setShowModal(!showModal);
+                }}
```
Comment (DanilWeda) [inline src/components/TodoList/TodoList.tsx:65]: You can write handler for it

File: src/components/TodoModal/TodoModal.tsx [hunk 13]
```diff
@@ -1,49 +1,80 @@
-import React from 'react';
+import React, { useContext } from 'react';
+// import { Loader } from '../Loader';
```
Comment (DanilWeda) [inline src/components/TodoModal/TodoModal.tsx:2]: remove commented

**Review by DanilWeda — CHANGES_REQUESTED** [review]
Good job! 
Check the loader comment and some small comments. 
Thanks!

**Review by olya-shyian — APPROVED** [review]
Great!🚀

**Pull request closed without merging**